src/main/java/baseball/Application.java

```diff
@@ -1,7 +1,164 @@
 package baseball;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.stream.Stream;
+
+import camp.nextstep.edu.missionutils.Console;
+import camp.nextstep.edu.missionutils.Randoms;
+
 public class Application {
-    public static void main(String[] args) {
-        //TODO: 숫자 야구 게임 구현
-    }
+	private final static String MESSAGE_END_GAME = "개의 숫자를 모두 맞히셨습니다! 게임 종료";
+	private final static String MESSAGE_BALL = "볼";
+	private final static String MESSAGE_STRIKE = "스트라이크";
+	private final static String MESSAGE_NOTHING = "낫싱";
+	private static final String REQUEST_INPUT_NUMBER = "숫자를 입력해주세요: ";
+	private static final String REQUEST_END_OR_RESTART = "게임을 새로 시작하려면 1, 종료하려면 2를 입력하세요";
+
+	private static final String COMMAND_RESTART_GAME = "1";
+	private static final String COMMAND_END_GAME = "2";
+	private static final int CONDITION_END = 3;
+	private static final int QUESTION_NUMBER_DIGITS = 3;
+	private static final char QUESTION_NUMBER_MIN = 1;
+	private static final char QUESTION_NUMBER_MAX = 9;
+
+	public static void main(String[] args) {
+		Application application = new Application();
+
+		while (true) {
+			List<Integer> answerNumber = application.makeAnswerNumber();
+
+			boolean isAnswer;
+			do {
+				int inputNumber = application.inputNumber();
+
+				isAnswer = application.compareNumber(answerNumber, inputNumber);
+			} while (isAnswer == false);
+
+			System.out.println(CONDITION_END + MESSAGE_END_GAME);
+
+			String inputStartOver = application.getInputStartOver();
+			if (application.isStartOver(inputStartOver) == false) {
+				break;
+			}
+		}
+	}
+
+	public List<Integer> makeAnswerNumber() {
+		List<Integer> answerNumber = new ArrayList<>();
+
+		while (answerNumber.size() < CONDITION_END) {
+			int randomNumber = Randoms.pickNumberInRange(1, 9);
+
+			if (answerNumber.contains(randomNumber) == false) {
+				answerNumber.add(randomNumber);
+			}
+
+		}
+
+		return answerNumber;
+	}
+
+	public int inputNumber() {
+		System.out.print(REQUEST_INPUT_NUMBER);
+		String inputNumber = Console.readLine();
+
+		handleException(inputNumber);
+
+		return Integer.parseInt(inputNumber);
+	}
+
+	public void handleException(String inputNumber) {
+		boolean isException = hasException(inputNumber);
+		if (isException) {
+			throw new IllegalArgumentException();
+		}
+	}
+
+	public boolean hasException(String inputNumber) {
+		TreeSet<Integer> inputNumberArray = new TreeSet<>();
```

Comment:
자바의 컬렉션 프레임워크(List, Set, Map)은 인터페이스에 핵심기능을 잘 정의한 대표적인 사례입니다. 현재 TreeSet 구현체에 있는 기능을 활용하는 게 아니기 때문에 확장을 위해 인터페이스 타입으로 할당해주세요!
그리고 Array라는 이름은 해당 변수의 자료형을 배열이라고 의심하게 만드므로, Set으로 명시해주는 게 좋겠네요.
```suggestion
		Set<Integer> inputNumberSet = new TreeSet<>();
```